cmd/acr/purge.go

```diff
@@ -463,43 +467,41 @@ func getManifestsToDelete(ctx context.Context, acrClient api.AcrCLIClientInterfa
 		return nil, err
 	}
 	// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
-	// that will not be deleted
-	doNotDelete := map[string]bool{}
-	candidatesToDelete := []acr.ManifestAttributesBase{}
-	// Iterate over all manifests to discover multiarchitecture manifests
+	// or the manifest has subjects attached
+	doNotDelete := make(Set[string])
```

Comment:
```suggestion
	doNotDelete := New[string]()
```
You may find that the function name `New` is not clear. It's because the `Set` type and the `New` function are not put int a right package or are not properly named.

Reply:
done.